File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/api/RepositoriesService.java
package com.mondiamedia.ahmedbadr.githubreopos.api;

import com.mondiamedia.ahmedbadr.githubreopos.models.GitRepo;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;

public interface RepositoriesService {

    @GET("/repositories")
    Observable<List<GitRepo>> getRepositories();
}

File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/api/RepositoriesService.kt
package com.mondiamedia.ahmedbadr.githubreopos.api

import com.mondiamedia.ahmedbadr.githubreopos.models.GitRepo

import retrofit2.http.GET

interface RepositoriesService {

    @GET("/repositories")
    suspend fun getRepositories(): List<GitRepo>
}
